File: src/main/java/io/github/et/subprocessLoader/ConfigLoader.java
package io.github.et.subprocessLoader;

import io.github.et.exceptions.BotInfoNotFoundException;

import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Properties;

public class ConfigLoader {
    public static ArrayList<MCServer> servers=new ArrayList<>();
    public static void load() throws IOException, BotInfoNotFoundException {
        File file=new File("./mcservers");
        if(!file.exists()){
            return;
        }
        for (File i:file.listFiles()){
            if(i.getName().toLowerCase().endsWith(".properties")){
                Properties a=new Properties();
                a.load(new FileReader(i));
                servers.add(new MCServer(a.getProperty("name"), a.getProperty("working_dir"), a.getProperty("command"), Long.parseLong(a.getProperty("group")), Boolean.parseBoolean(a.getProperty("useBackup")),Boolean.parseBoolean(a.getProperty("death_msg")), Boolean.parseBoolean(a.getProperty("useAdvancement"))));
            }
        }
    }
}

File: SubProcess/src/main/kotlin/io/github/et/ConfigLoader.kt
package io.github.et

import java.io.File
import java.io.FileReader
import java.io.IOException
import java.nio.charset.StandardCharsets
import java.util.*

object ConfigLoader {
    @JvmField
    var servers: ArrayList<MCServer> = ArrayList()

    @JvmStatic
    @Throws(IOException::class)
    fun load() {
        val root=File("./mcservers")
        if(!root.exists()){
            return
        }
        for (i in root.listFiles()!!) {
            if(i.name.lowercase(Locale.getDefault()).endsWith(".properties")){
                val a =Properties()
                a.load(FileReader(i, StandardCharsets.UTF_8))
                servers.add(MCServer(a["name"].toString(), a["working_dir"].toString(),a["command"].toString(),a["group"].toString().toLong(),a["useBackup"].toString().toBoolean(),a["death_msg"].toString().toBoolean() ,a["useAdvancement"].toString().toBoolean()))
            }
        }

    }
}
